File: chroma-league/src/test/java/com/bonepl/chromaleague/hud/parts/resource/KledCourageBarTest.java
package com.bonepl.chromaleague.hud.parts.resource;

import com.bonepl.chromaleague.IntSteps;
import com.bonepl.chromaleague.hud.AnimationTester;
import org.junit.jupiter.api.Test;

class KledCourageBarTest extends AbstractResourceTest {

    @Test
    void testKledCourageBar() {
        final IntSteps intSteps = new IntSteps(0, 100, 5);
        final KledCourageBar kledCourageBar = new KledCourageBar();
        new AnimationTester()
                .withBeforeIterationAction(i -> {
                    if (i >= 20 && i <= 50) {
                        mockResource(100, 100);
                    } else {
                        mockResource(intSteps.nextInt(), 100);
                    }
                })
                .testAnimation(kledCourageBar, 70);
    }
}

File: test/hud/parts/resource/KledCourageBar.test.ts
import {describe, test} from 'vitest';
import {IntSteps} from '../../../IntSteps.js';
import {AnimationTester} from '../../../AnimationTester.js';
import {KledCourageBar} from '../../../../src/hud/parts/resource/KledCourageBar.js';
import {mockResource, setupResourceTest} from './AbstractResourceTest.js';

/** Integration test for Kled courage resource bar on actual hardware. */
describe('KledCourage Integration', () => {
    setupResourceTest();

    test('testKledCourageBar', async () => {
        const intSteps = new IntSteps(0, 100, 5);
        const kledCourageBar = new KledCourageBar();

        await new AnimationTester()
            .withBeforeIterationAction(i => {
                if (i >= 20 && i <= 50) {
                    mockResource(100, 100);
                } else {
                    mockResource(intSteps.nextInt(), 100);
                }
            })
            .testAnimation(kledCourageBar, 70);
    }, 30000);
});
